src/backsolve_adjoint.jl

```diff
@@ -215,9 +215,14 @@ end
 
 
 
-function backsolve_checkpoint_callbacks(sensefun, sol, checkpoints, callback)
+function backsolve_checkpoint_callbacks(sensefun, sol, checkpoints, callback, duplicate_iterator_times=nothing)
   prob = sol.prob
-  cur_time = Ref(length(checkpoints))
+  if duplicate_iterator_times !== nothing
+    _checkpoints = filter(x->x ∉ duplicate_iterator_times[1], checkpoints)
```

Comment:
The callbacks in Backsolve didn't work because ultimately the wrong state u entered into the reverse callback.  This filters the duplicates fully out of the checkpointing. If we want to keep a checkpoint at the affect times we should somehow make sure that the checkpoint always refers to the proper limit.

Reply:
I think removing the event times is a good idea.